pwiz_tools/Skyline/Model/Lib/AlphaPeptDeep/AbstractDeepLibraryBuilder.cs

```diff
@@ -255,28 +526,22 @@ public IList<string> GetWarningMods()
                     var mod = modifiedSequence.ExplicitMods[i];
                     if (!mod.UnimodId.HasValue)
                     {
-                        var haveMod = resultList.FirstOrDefault(m => m == mod.Name);
-                        if (haveMod == null)
-                        {
-                            resultList.Add(mod.Name);
-                        }
+                        _warningModSupports[mod.Name] = PredictionSupport.NONE;
                     }
-
-                    var unimodIdWithName = mod.UnimodIdWithName;
-                    var modNames = ModificationTypes.Where(m => m.Accession == unimodIdWithName).ToArray();
-                    if (modNames.Length == 0)
+                    else
                     {
-                        var haveMod = resultList.FirstOrDefault(m => m == mod.Name);
-                        if (haveMod == null)
-                        {
-                            resultList.Add(mod.Name);
-                        }
+                        if (!(libraryBuilderModificationSupport.IsMs2SupportedMod(mod.UnimodId) &&
```

Comment:
I think you should make public the `GetModificationType` method on `LibraryBuilderModificationSupport`.

Then you would be able to do:
```
var support = libraryBuilderModificationSupport.GetModificationType(mod.UnimodId.Value)?.SupportedModels ?? PredictionSupport.NONE;
if (!Equals(support, PredictionSupport.ALL))
{
    _warningModSupports[mod.Name] = support;
}
```

I don't think the LibraryBuildModificationSupport class needs all the variations of "IsMs2SupportedMod", "IsRtSupportedMod" etc.
If you make it really easy to get a PredictionSupport from a unimod ID, the caller will be able to do things more efficiently.